proto_test.go

```diff
@@ -368,10 +368,59 @@ func BenchmarkDecodeMessageBig(b *testing.B) {
 	}
 }
 
+func TestEncodeSmallMessage(t *testing.T) {
+	buf := new(bytes.Buffer)
+	err := smallMessage.EncodeTo(buf, binary.LittleEndian)
+	if err != nil {
+		t.Fatal(err)
+	}
+	expected := [][]byte{
```

Comment:
these binary outputs should be moved to testdata files that are read during test runtime